crates/ruff_linter/src/rules/flake8_pyi/rules/unused_private_type_definition.rs

```diff
@@ -185,13 +192,26 @@ pub(crate) fn unused_private_type_var(
         let Expr::Call(ast::ExprCall { func, .. }) = value.as_ref() else {
             continue;
         };
-        if !checker.semantic().match_typing_expr(func, "TypeVar") {
+
+        let semantic = checker.semantic();
+        let Some(type_var_like_kind) = semantic.resolve_call_path(func).and_then(|call_path| {
+            if semantic.match_typing_call_path(&call_path, "TypeVar") {
+                Some("TypeVar")
+            } else if semantic.match_typing_call_path(&call_path, "ParamSpec") {
+                Some("ParamSpec")
+            } else if semantic.match_typing_call_path(&call_path, "TypeVarTuple") {
```

Comment:
@AlexWaygood - I made one change here per your comment in the summary. Each call to `semantic.match_typing_expr` does a lookup internally to map the expression to a "call path", like `["typing", "TypeVar"]`. However, we can just do that lookup once via `semantic.resolve_call_path`, then pass the call path to `semantic.match_typing_call_path`. (`semantic.match_typing_expr` is just a wrapper around this process, so if we want to do multiple checks, it's more efficient to do the lookup outside of the `semantic.match_typing_expr`.)

Reply:
Ahh nice, thanks! Yep, that sounds like it's exactly what I was looking for :)